libraries/tipipeline/vars/component.groovy

```diff
@@ -96,6 +96,136 @@ def checkoutV2(gitUrl, keyInComment, prTargetBranch, prCommentBody, credentialsI
+def checkoutSupportBatch(gitUrl, keyInComment, prTargetBranch, prCommentBody, refs, credentialsId="", trunkBranch="master", timeout=5) {
+    def tidbTestRefs = [] // List of tidb-test refs PR:123, PR:456
+    boolean branchOrCommitSpecified = false // Flag to check if a branch or commit is specified in any PR title
+
+    refs.pulls.each { pull ->
+        def componentBranch = computeBranchFromPR(keyInComment, prTargetBranch, pull.title,  trunkBranch)
+        if (componentBranch.startsWith("pr/")) {
+            tidbTestRefs.add("PR:${componentBranch}") // Add as PR reference
+        } else {
+            // some PR title contains a branch or commit
+            if ( prTargetBranch != componentBranch) {
+                branchOrCommitSpecified = true
+                tidbTestRefs.add("Branch:${componentBranch}") // Add as branch reference specified in PR title
+            }
+        }
+    }
+    
+    if (tidbTestRefs.isEmpty()) {
+        echo "No tidb-test refs specified, defaulting to base branch ${prTargetBranch} of tidb-test."
+        checkoutSingle(gitUrl, prTargetBranch, prTargetBranch, credentialsId)
+    } else if (tidbTestRefs.size() == 1 && tidbTestRefs[0].startsWith("Branch:")) {
+        // default branch or specific branch
+        // Single PR with branch specified
+        echo "Single PR with tidb-test branch specified: ${prTargetBranch}"
+        def branch = tidbTestRefs[0].split(":")[1]
+        checkoutSingle(gitUrl, prTargetBranch, branch, credentialsId)
+    // if tidbTestRefs size > 1 and any of tidbTestRefs start with branch , then error and exit
+    } else if (tidbTestRefs.size() > 1 && branchOrCommitSpecified) {
+        echo "Error: Specifying a tidb-test branch is not supported for multiple tidb PRs."
+        throw new Exception("Error: Specifying a tidb-test branch is not supported for multiple tidb PRs batch.")
+    } else {
+        // multi PR specified PR (notice: for batch merge with specific branch is not supported)
+        // single PR with specified PR 
+        checkoutPRWithPreMerge(gitUrl, prTargetBranch, tidbTestRefs.collect { it.split(":")[1] } as List, credentialsId)
+    }
+}
+
+def checkoutSingle(gitUrl, prTargetBranch, branchOrCommit, credentialsId, timeout=5) {
+    def refSpec = "+refs/heads/*:refs/remotes/origin/*"
+    checkout(
+        changelog: false,
+        poll: true,
+        scm: [
+            $class: 'GitSCM',
+            branches: [[name: branchOrCommit]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [
+                [$class: 'PruneStaleBranch'],
+                [$class: 'CleanBeforeCheckout'],
+                [$class: 'CloneOption', timeout: timeout],
+            ], 
+            submoduleCfg: [],
+            userRemoteConfigs: [[
+                credentialsId: credentialsId,
+                refspec: refSpec,
+                url: gitUrl,
+            ]]
+        ]
+    )
+}
```

Comment:
It's better to unify the usage for git cloning and checkout, please create a backlog issue task